javascripts/discourse/components/localized-header.js

```diff
@@ -16,6 +16,11 @@ export default Component.extend({
       (obj) => obj.locale === I18n.currentLocale().replace(/_/g, "-")
     );
 
+    // remove special chars, spaces, from link class
+    filteredLocale[0].links.forEach((link) => {
+      link.link_class = dasherize(link.link_text.replace(/[^a-zA-Z]/, ""));
+    });
+
```

Comment:
This should be moved below that next `if`, otherwise if no locale is set it will fail.